mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -293,23 +299,41 @@ static AllocOp allocAndInitializeTensor(OpBuilder &builder, Block *block,
       builder.getUnknownLoc(), cpuAllocOp, fourDimUnknownSizeMemRefType);
   block->push_back(cpuMemRefCastOp);
 
-  // Populate initial values of the output tensor
-  auto cpuMemsetOp =
-      builder.create<CallOp>(builder.getUnknownLoc(), mcpuMemset4DFuncOp,
-                             ValueRange{cpuMemRefCastOp, memsetValue});
+  // Populate initial values
+  auto cpuMemsetOp = builder.create<CallOp>(
+      builder.getUnknownLoc(), mcpuMemset4DFuncOp,
+      ValueRange{cpuMemRefCastOp, memsetMinValue, memsetMaxValue, seedValue});
   block->push_back(cpuMemsetOp);
 
   return cpuAllocOp;
 }
 
-static FuncOp createConvertResult(ModuleOp &module, OpBuilder &builder,
-                                  mlir::MemRefType &originalMemRefType,
-                                  mlir::MemRefType &convertedMemRefType) {
-  // Create print_result function
+static FuncOp
+createConvertResult(ModuleOp &module, OpBuilder &builder,
+                    mlir::MemRefType &originalMemRefType,
+                    mlir::MemRefType &convertedMemRefType,
+                    std::unordered_map<std::string, FuncOp> &convertFuncs) {
+  std::string funcName = "convert_tensor";
+  auto originalMemRefShape = originalMemRefType.getShape();
+  funcName += std::to_string(originalMemRefShape[0]);
+  for (unsigned i = 1; i < originalMemRefShape.size(); ++i) {
+    auto dim = originalMemRefShape[i];
+    funcName += "x" + std::to_string(dim);
+  }
+
```

Comment:
I need to convert fp16 to fp32 in order to copy a fp16 tensor to a cpu convolution tensor. So change the funcName to "convert_tensor" followed by sizes of the tensor, e.g., convert_tensor128x3x3x8(), to avoid duplicated function declarations.